### DISCUSSION TITLE
MonoGame Android Project in VS cannot be <OutputType>Exe</OutputType> ?

### DISCUSSION THREAD

**Discussion by Logopher** [MonoGame/MonoGame#8268] [Android]
I made a new "MonoGame Android Application" and had to set it to use .Net 8 and target the correct versions of Android, and then it built. And then it refused to run. Poking at it, I discovered that a .Net 8 Console Application which is set to use Android as its platform promptly becomes a Class Library. I even have my .csproj specifying {OutputType}Exe{/OutputType} and the StartupObject, and msbuild (run directly) insists on passing /target:library to csc which then gets angry at me that I specified /main (the StartupObject).

What gives?

**Comment by Logopher** [comment]
This was resolved in Discord. Just remove the StartupObject element and tolerate the "Class Library" status of the project. It will build and run anyway.

**Reply by dellis1972** [reply]
This is by design for .NET Android. The OutputType will always end up as `dll` because you cannot run `exe` on android. The assemblies are packaged up into an apk or aab.  The StartUpObject is also redundant as it is never used.
